sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs

```diff
@@ -1996,6 +1998,461 @@ internal virtual void SetNameFieldsIfNull()
+
+                        cancellationToken.ThrowIfCancellationRequested();
+                    } while (!string.IsNullOrEmpty(continuationToken));
+
+                    return new ChangeAccessControlListResult()
+                        {
+                            DirectoriesSuccessfulCount = directoriesSuccessfulCount,
+                            FilesSuccessfulCount = filesSuccessfulCount,
+                            FailureCount = failureCount,
+                        };
+                }
+                catch (Exception ex)
+                {
+                    Pipeline.LogException(ex);
+                    throw;
+                }
+                finally
+                {
+                    Pipeline.LogMethodExit(nameof(DataLakePathClient));
+                }
+            }
+        }
+
+        private static SetAccessControlRecursiveResponse DeserializeSetAccessControlRecursiveResponse(JsonElement element)
+        {
+            SetAccessControlRecursiveResponse result = new SetAccessControlRecursiveResponse();
+            foreach (var property in element.EnumerateObject())
+            {
+                if (property.NameEquals("directoriesSuccessful"))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        continue;
+                    }
+                    result.DirectoriesSuccessful = property.Value.GetInt32();
+                    continue;
+                }
+                if (property.NameEquals("filesSuccessful"))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        continue;
+                    }
+                    result.FilesSuccessful = property.Value.GetInt32();
+                    continue;
+                }
+                if (property.NameEquals("failureCount"))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        continue;
+                    }
+                    result.FailureCount = property.Value.GetInt32();
+                    continue;
+                }
+                if (property.NameEquals("failedEntries"))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        continue;
+                    }
+                    var failedEntries = new List<AclFailedEntryList>();
+                    foreach (var item in property.Value.EnumerateArray())
+                    {
+                        failedEntries.Add(DeserializeAclFailedEntryList(item));
+                    }
+                    result.FailedEntries = failedEntries;
+                    continue;
+                }
+            }
+            return result;
+        }
+
+        private static AclFailedEntryList DeserializeAclFailedEntryList(JsonElement element)
```

Comment:
If we have to end up keeping the json work around, let's move this to a different file.  Maybe ```DataLakeExtensions.cs```?

Reply:
moved.